src/main/java/ognl/DefaultClassResolver.java

```diff
@@ -41,28 +42,30 @@
  */
 public class DefaultClassResolver extends Object implements ClassResolver
 {
-    private Map     classes = new HashMap(101);
+    private Map     classes = new ConcurrentHashMap(101);
```

Comment:
Can you use generics here? This will reduce casting below.

Reply:
I've changed to use generics.